libpod/runtime.go

```diff
@@ -565,6 +566,33 @@ func (r *Runtime) Info() ([]InfoData, error) {
 	return info, nil
 }
 
+func (r *Runtime) GenerateName() (string, error) {
+	var name string
+	ctrs, err := r.state.AllContainers()
+	if err != nil {
+		return "", errors.Wrapf(err, "error retrieving all containers from state")
+	}
+	pods, err := r.state.AllPods()
```

Comment:
AllContainers() and AllPods() are very expensive operations - it would probably be better to do individual LookupContainer and LookupPod queries for each name.

Reply:
That's a much better solution.